client/tests/simulator_tests.rs

```diff
@@ -83,6 +83,63 @@ fn test_get_capability_manufacturer_id() {
     assert_eq!(received, expected);
 }
 
+#[test]
+fn test_get_random() {
+    const REQUEST_SIZE: u16 = 32;
+    const MIN_RECEIVED: u16 = 32;
+
+    let (_sim_lifeline, mut tpm) = get_started_tpm();
+
+    let command = GetRandomCmd {
+        bytes_requested: REQUEST_SIZE,
+    };
+
+    // We allow panic in test cases.
+    let resp = run_command(&command, &mut tpm).expect("Failed running command.");
+
+    // Validate that retrieved data contains at least 32 values (SHA256 digest),
+    // and is less that requested amount.
+    assert!(resp.random_bytes.size >= MIN_RECEIVED);
+    assert!(resp.random_bytes.size <= REQUEST_SIZE);
+
+    // Lets pull out the actual data as a slice for convenience
+    let random_slice = &resp.random_bytes.buffer[0..resp.random_bytes.size as usize];
+
+    // Print for test inspection.
+    println!("Got random sequence: {:?}", random_slice);
+
+    // Crude randomness test. Confirm we did not get identical data sequences.
+    {
+        let mut zeros: u16 = 0;
+        let mut count: [u16; 16] = [0; 16]; // 16 slots.
+        let mut last_value = random_slice[random_slice.len() - 1];
+        let mut i = 0;
+        for i in 0..random_slice.len() as usize {
+            // Calculate the circular distance modulo 256 going upwards from first value.
+            // This will give an even probability distribution of values [0 ; 255].
+            let diff: u16 = (256 as u16 + random_slice[i] as u16 - last_value as u16) % 256;
```

Comment:
I assume this `256` is to avoid underflow, but personally I'd just embrace it and use `u8` for your logic here: https://doc.rust-lang.org/std/primitive.u8.html#method.wrapping_sub

Reply:
I have simplified the test - so construct no longer used 👍